lib/ansible/modules/network/cnos/cnos_vlan.py

```diff
@@ -223,66 +215,39 @@ def main():
             vlanArg5=dict(required=False),),
         supports_check_mode=False)
 
-    username = module.params['username']
-    password = module.params['password']
-    enablePassword = module.params['enablePassword']
-    vlanArg1 = module.params['vlanArg1']
-    vlanArg2 = module.params['vlanArg2']
-    vlanArg3 = module.params['vlanArg3']
-    vlanArg4 = module.params['vlanArg4']
-    vlanArg5 = module.params['vlanArg5']
     outputfile = module.params['outputfile']
-    hostIP = module.params['host']
-    deviceType = module.params['deviceType']
 
     output = ""
-    if not HAS_PARAMIKO:
-        module.fail_json(msg='paramiko is required for this module')
-
-    # Create instance of SSHClient object
-    remote_conn_pre = paramiko.SSHClient()
-
-    # Automatically add untrusted hosts (make sure okay for security policy in
-    # your environment)
-    remote_conn_pre.set_missing_host_key_policy(paramiko.AutoAddPolicy())
-
-    # initiate SSH connection with the switch
-    remote_conn_pre.connect(hostIP, username=username, password=password)
-    time.sleep(2)
-
-    # Use invoke_shell to establish an 'interactive session'
-    remote_conn = remote_conn_pre.invoke_shell()
     time.sleep(2)
 
     # Enable and enter configure terminal then send command
-    output = output + cnos.waitForDeviceResponse("\n", ">", 2, remote_conn)
-
-    output = output + \
-        cnos.enterEnableModeForDevice(enablePassword, 3, remote_conn)
+    cmd = [{'command': '\n', 'prompt': None, 'answer': None}]
+    output = output + str(cnos.run_cnos_commands(module, cmd))
 
     # Make terminal length = 0
-    output = output + \
-        cnos.waitForDeviceResponse("terminal length 0\n", "#", 2, remote_conn)
+    cmd = [{'command': 'terminal length 0', 'prompt': None, 'answer': None}]
+    output = output + str(cnos.run_cnos_commands(module, cmd))
 
     # Go to config mode
-    output = output + \
-        cnos.waitForDeviceResponse("configure device\n", "(config)#", 2, remote_conn)
+    cmd = [{'command': 'configure terminal', 'prompt': None, 'answer': None}]
+    output = output + str(cnos.run_cnos_commands(module, cmd))
+    if(output.find('Error') != -1):
+        cmd = [{'command': 'configure device', 'prompt': None, 'answer': None}]
+        output = output + str(cnos.run_cnos_commands(module, cmd))
 
     # Send the CLi command
-    output = output + \
-        cnos.vlanConfig(
-            remote_conn, deviceType, "(config)#", 2, vlanArg1, vlanArg2,
-            vlanArg3, vlanArg4, vlanArg5)
+    output = output + str(cnos.vlanConfig(module, "(config)#", None))
 
-    # Save it into the file
+    # Save it operation details into the file
     file = open(outputfile, "a")
     file.write(output)
     file.close()
 
     # need to add logic to check when changes occur or not
     errorMsg = cnos.checkOutputForError(output)
```

Comment:
This _should_ be unnecessary, assuming you have added all applicable error responses to `terminal_stderr_re` in the terminal plugin. `network_cli` uses those values to automatically determine if an error has occurred and report the error message for you.

Reply:
The legacy code is in the style of an NMS system we have. Here we have command-rules file and error code file in utils. Let me explain, Suppose a person inputs vlan id as 5000 whose upper range  is 4095. This value is checked against a rule file pertaining to the type of device and the error is thrown upfront to him as Error Code and String. So CLI wont get executed to get error and display it to user.  Input validations are done against type, range and value of the data inputs.  Still device may go through errors depending on some original settings prevalent there. Such cases only will come in  terminal_stderr_re. Based on the present "Pro-active" style adopted in the code, this method is required. In the legacy code, these errors are shown not only in screen, but in file too